app/controllers/manage_metadata_controller.rb

```diff
@@ -57,6 +57,61 @@ def set_variable_information
     end
   end
 
+  def set_service
+    @concept_id = params[:service_id] || params[:id]
+    @revision_id = params[:revision_id]
+
+    # retrieve the variable metadata with the current umm_var version
+    headers = { 'Accept' => "application/#{Rails.configuration.umm_var_version}; charset=utf-8" }
+    service_concept_response = cmr_client.get_concept(@concept_id, token, headers, @revision_id)
+
+    @service = if service_concept_response.success?
+                  service_concept_response.body
+                else
+                  Rails.logger.error("Error retrieving concept for Variable #{@concept_id} in `set_service`: #{service_concept_response.inspect}")
+                  {}
+                end
+
+    set_service_information
+  end
+
+  def set_service_information
+    # search for service by concept id to get the native_id and provider_id
+    # if the service is not found, try again because CMR might be a little slow to index if it is a newly published record
+    attempts = 0
+    while attempts < 20
+      services_search_response = cmr_client.get_services(concept_id: @concept_id, all_revisions: true)
+
+      service_data = if services_search_response.success?
+                       services_search_response.body.fetch('items', [])
+                     else
+                       []
+                     end
+      service_data.sort! { |a, b| b['meta']['revision-id'] <=> a['meta']['revision-id'] }
+
+      @revisions = service_data
+      latest = service_data.first
+      meta = latest.nil? ? {} : latest.fetch('meta', {})
+
+      if !@revision_id.nil? && meta['revision-id'].to_s != @revision_id.to_s
```

Comment:
I think this might introduce a bug we have for variables where after publishing a new revision, you still might see a banner saying you are viewing an old revision. MMT-1186. I _think_ the solution might just be to add an `else` statement to set the ivar to false